client/components/domains/domain-registration-suggestion/index.jsx

```diff
@@ -132,6 +132,109 @@ class DomainRegistrationSuggestion extends Component {
+		} );
+
+		switch ( priceRule ) {
+			case 'ONE_TIME_PRICE':
+				return translate( '%(baseLabel)s. %(price)s one-time', {
+					args: {
+						baseLabel,
+						price: productCost,
+					},
+					comment:
+						'Accessible label for one-time priced domain (e.g. domain with 100-year plan). %(baseLabel)s is the base label (e.g. "Select domain testing.com"). %(price)s is the price.',
+				} );
+			case 'FREE_DOMAIN':
+				return translate( '%(baseLabel)s. Free', {
+					args: {
+						baseLabel,
+					},
+					comment:
+						'Accessible label for free domain. %(baseLabel)s is the base label (e.g. "Select domain testing.com").',
+				} );
+			case 'FREE_FOR_FIRST_YEAR':
+				return translate( '%(baseLabel)s. Free for the first year, then %(price)s per year', {
+					args: {
+						baseLabel,
+						price: productCost,
+					},
+					comment:
+						'Accessible label for domain free for the first year. %(baseLabel)s is the base label (e.g. "Select domain testing.com"). %(price)s is the price.',
+				} );
+			case 'FREE_WITH_PLAN':
+				return translate(
+					'%(baseLabel)s. Free for the first year with annual paid plans, then %(price)s per year',
+					{
+						args: {
+							baseLabel,
+							price: productCost,
+						},
+						comment:
+							'Accessible label for free domain with normal price. %(baseLabel)s is the base label (e.g. "Select domain testing.com"). %(price)s is the price.',
+					}
+				);
+			case 'UPGRADE_TO_HIGHER_PLAN_TO_BUY':
+				return translate( '%(baseLabel)s. Plan upgrade required to register this domain.', {
+					args: {
+						baseLabel,
+					},
+					comment:
+						'Accessible label for domain that requires a plan upgrade. %(baseLabel)s is the base label (e.g. "Select domain testing.com").',
+				} );
+			case 'INCLUDED_IN_HIGHER_PLAN':
+				return translate( '%(baseLabel)s. Included in paid plans', {
+					args: {
+						baseLabel,
+					},
+					comment:
+						'Accessible label for domain included in higher plans. %(baseLabel)s is the base label (e.g. "Select domain testing.com").',
+				} );
+			case 'DOMAIN_MOVE_PRICE':
+				return translate( '%(baseLabel)s. %(price)s one-time', {
+					args: {
+						baseLabel,
+						price: productCost,
+					},
+					comment:
+						'Accessible label for domain move price. %(baseLabel)s is the base label (e.g. "Select domain testing.com"). %(price)s is the price.',
+				} );
+			case 'PRICE':
+				if ( productSaleCost && productCost ) {
+					return translate(
+						'%(baseLabel)s. %(salePrice)s for the first year, then %(price)s per year',
+						{
+							args: {
+								baseLabel,
+								salePrice: productSaleCost,
+								price: productCost,
+							},
+							comment:
+								'Accessible label for domain with sale price. %(baseLabel)s is the base label (e.g. "Select domain testing.com"). %(salePrice)s is the sale price. %(price)s is the price.',
+						}
+					);
+				} else if ( productCost ) {
```

Comment:
Nit: Since we're returning in this `else if` (and in the `if` above), we could drop the `else` and use only an `if` here. It helps a bit with readability. Super minor and definitely a matter of style! 😄